### ISSUE TITLE
do we have a trafo_world_to_cam for each image or it remains fixed?

### ISSUE THREAD

**Issue by Rajat-Mehta** [thomasfermi/Algorithms-for-Automated-Driving#10]
I see in your jupyter noteback that you have a trafo_world_to_cam file with same name as the input image. Does that mean we have a different transformation file for each image or we can reuse the same file (obviously taken from same camera under same settings) ?

Thanks in advance.

**Comment by thomasfermi** [comment]
Hi  @Rajat-Mehta ! I do not know which Jupyter Notebook you mean, but I guess I can answer your question anyway.

In general, different images are captured from different positions (and orientations).  This is because the camera is attached to a vehicle that is moving. Hence, there is a different transformation "world_to_cam" for each image. The camera is moving with respect to the stationary world coordinate system.

TLDR: You cannot reuse the transformation files. Each image needs its own transformation matrix.

**Comment by thomasfermi** [comment]
Hi @Rajat-Mehta! I think you mixed up some parts of the book.

The `trafo_world_to_cam` matrices are only needed in the part of the book where we create training data for the neural network. We get the lane boundaries in the world frame from Carla's API. We also get the `trafo_world_to_cam ` from Carla. We combine this to get the lane boundaries in the camera reference frame. Then we project those lane boundaries into a "label image". This is used to train the neural net. In the real world, you could get these label images by manually drawing lane lines into your image.

Now to this point:
> ... convert the lane detection results from pixel to world (or road) coordinate system.

There is a big between what I called the "road frame" and what I called the "world frame" in the book. Please have a look at [this image](https://thomasfermi.github.io/Algorithms-for-Automated-Driving/_images/coordinate_systems.svg) from the book again. The world frame is stationary (or fixed to planet earth if you will). The road frame however is fixed to the car. It is moving with the car. Hence, the `trafo_road_to_cam ` is constant. You will see this if you do the exercises on lane detection where you implement all the #TODOs in [camera_geometry.py](https://github.com/thomasfermi/Algorithms-for-Automated-Driving/blob/master/code/exercises/lane_detection/camera_geometry.py). One #TODO asks you to set the proper value for `self.trafo_cam_to_road`. This matrix is constant.
